certbot/certbot/_internal/plugins/manual.py

```diff
@@ -114,11 +124,11 @@ def get_chall_pref(self, domain):
 
     def perform(self, achalls):  # pylint: disable=missing-function-docstring
         responses = []
-        for achall in achalls:
+        for i, achall in enumerate(achalls, 1):
```

Comment:
So this prints the message on the very last challenge, which may or may not be a DNS challenge. If we sort `achalls` by type so that DNS challs come last before calling perform, this will work, but otherwise it might not get printed at all. Alternatively, we could count the number of DNS challenges beforehand, then set the flag to true when we get to the final dns challenge.

Reply:
Hmm, I didn't know it was possible to mix different challenge types with the `manual` plugin. I'll look into it.

Reply:
I don't really like iterating over `achalls` *twice*, but I didn't know another way and this was fairly easy :stuck_out_tongue: